rasa/engine/graph.py

```diff
@@ -22,40 +22,44 @@
 
 @dataclass
 class SchemaNode:
-    """Represents one node in the schema."""
+    """Represents one node in the schema.
+
+    Args:
+        needs: describes which parameters in `fn` (or `constructor_name`
+            if `eager==False`) are filled by which parent nodes.
+        uses: The class which models the behavior of this specific graph node.
+        constructor_name: The name of the constructor which should be used to
+            instantiate the component. If `eager==False` then the `constructor` can
+            also specify parameters which are filled by parent nodes. This is e.g.
+            useful if a parent node returns a `Resource` and this node wants to
+            directly load itself from this resource.
+        fn: The name of the function which should be called on the instantiated
+            component when the graph is executed. The parameters from `needs` are
+            filled from the parent nodes.
+        config: The configuration for this graph node.
```

Comment:
I wonder if we shouldn't change this to mention that this is the config specified by the user, which will then be merged with the default config?